src/backend/cuda/nvrtc/cache.cpp

```diff
@@ -89,6 +96,49 @@ void Kernel::setConstant(const char* name, CUdeviceptr src, size_t bytes) {
     CU_CHECK(cuMemcpyDtoDAsync(dst, src, bytes, getActiveStream()));
 }
 
+pair<int, int> getComputeFlag() {
+    struct cuNVRTCcompute {
+        /// The CUDA Toolkit version returned by cudaRuntimeGetVersion
+        int cuda_version;
+
+        /// Maximum major compute flag supported by cuda_version
+        int major;
+
+        /// Maximum minor compute flag supported by cuda_version
+        int minor;
+    };
+    static const cuNVRTCcompute Toolkit2Compute[] = {
+        {10010, 7, 5},
```

Comment:
I think you should convert the compute versions to int format on the right. There is not maximum minor version. For example if the maximum version was 7.0, then the condition below will fail for version 5.3. You can use the https://github.com/arrayfire/arrayfire/blob/bce559f881006932de7c2a6787217dacc4c52937/src/backend/cuda/platform.cpp#L246 to convert it to string.